packages/material-ui-lab/src/TreeView/TreeView.js

```diff
@@ -239,19 +263,22 @@ const TreeView = React.forwardRef(function TreeView(props, ref) {
    * Focus Helpers
    */
 
-  const focus = (id) => {
+  const focus = (event, id) => {
     if (id) {
-      setTabbable(id);
       setFocusedNodeId(id);
+
+      if (onNodeFocus) {
+        onNodeFocus(event, id);
```

Comment:
I'm not a big fan of passing an event here. Sometimes the target will be the tree e.g. on keyboard events.

Reply:
This is fine in my book as long as it's typed as a generic synthetic event i.e. `React.SyntheticEvent`. We already have this in place in a couple of other places. It's similar to how React itself doesn't make any guarantee about `event.nativeEvent`. https://github.com/mui-org/material-ui/pull/21552 has more context on this issue.